connector/connect/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectDeduplicateSuite.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.connect.planner
+
+import org.apache.spark.sql.{Dataset, Row, SparkSession}
+import org.apache.spark.sql.catalyst.expressions.AttributeReference
+import org.apache.spark.sql.test.SharedSparkSession
+import org.apache.spark.sql.types.{IntegerType, StringType, StructField, StructType}
+
+/**
+ * [[SparkConnectPlanTestWithSparkSession]] contains a SparkSession for the connect planner.
+ *
+ * It is not recommended to use Catalyst DSL along with this trait because `SharedSparkSession`
```

Comment:
This makes me think that we probably shouldn't use catalyst DSL at all. The tests need Spark Connect planner, which needs `SparkSession`. Some tests happen to not invoke `SparkSession`, but it's a bit hacky to rely on this assumption. We should just compare plans produced by proto DSL and DataFrame APIs.

Reply:
We decided to with Catalyst DSL in https://github.com/apache/spark/pull/37994. However that also caused the pain of does the small scope to avoid implicit conflicts.

Given that we need to need a session based test, migrate all such tests to the same place that uses session and also DataFrame API just makes sense.  Also I believe in this approach we don't have implicit conflict anymore in the same scope.

However about after this PR, let me send a follow up PR for the testing refactoring?

Reply:
We can do it in a followup.